src/crimson/osd/ops_executer.cc

```diff
@@ -483,184 +483,197 @@ OpsExecuter::execute_op(OSDOp& osd_op)
-    }, true);
-
-  // watch/notify
-  case CEPH_OSD_OP_WATCH:
-    return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
-      return do_op_watch(osd_op, os, txn);
-    }, false);
-  case CEPH_OSD_OP_LIST_WATCHERS:
-    return do_read_op([this, &osd_op] (auto&, const auto& os) {
-      return do_op_list_watchers(osd_op, os);
-    });
-  case CEPH_OSD_OP_NOTIFY:
-    return do_read_op([this, &osd_op] (auto&, const auto& os) {
-      return do_op_notify(osd_op, os);
-    });
-  case CEPH_OSD_OP_NOTIFY_ACK:
-    return do_read_op([this, &osd_op] (auto&, const auto& os) {
-      return do_op_notify_ack(osd_op, os);
-    });
-  case CEPH_OSD_OP_ASSERT_VER:
-    return do_read_op([this, &osd_op] (auto&, const auto& os) {
-      return do_assert_ver(osd_op, os);
-    });
+      return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
+        return backend.omap_remove_range(os, osd_op, txn, delta_stats);
+      }, true);
+    case CEPH_OSD_OP_OMAPRMKEYS:
+      /** TODO: Implement supports_omap()
+      if (!pg.get_pool().info.supports_omap()) {
+        return crimson::ct_error::operation_not_supported::make();
+      }*/
+      return do_write_op([&osd_op] (auto& backend, auto& os, auto& txn) {
+        return backend.omap_remove_key(os, osd_op, txn);
+      }, true);
+    case CEPH_OSD_OP_OMAPCLEAR:
+      return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
+        return backend.omap_clear(os, osd_op, txn, *osd_op_params, delta_stats);
+      }, true);
+
+    // watch/notify
+    case CEPH_OSD_OP_WATCH:
+      return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
+        return do_op_watch(osd_op, os, txn);
+      }, false);
+    case CEPH_OSD_OP_LIST_WATCHERS:
+      return do_read_op([this, &osd_op] (auto&, const auto& os) {
+        return do_op_list_watchers(osd_op, os);
+      });
+    case CEPH_OSD_OP_NOTIFY:
+      return do_read_op([this, &osd_op] (auto&, const auto& os) {
+        return do_op_notify(osd_op, os);
+      });
+    case CEPH_OSD_OP_NOTIFY_ACK:
+      return do_read_op([this, &osd_op] (auto&, const auto& os) {
+        return do_op_notify_ack(osd_op, os);
+      });
+    case CEPH_OSD_OP_ASSERT_VER:
+      return do_read_op([this, &osd_op] (auto&, const auto& os) {
+        return do_assert_ver(osd_op, os);
+      });
 
-  default:
-    logger().warn("unknown op {}", ceph_osd_op_name(op.op));
-    throw std::runtime_error(
-      fmt::format("op '{}' not supported", ceph_osd_op_name(op.op)));
-  }
+    default:
+      logger().warn("unknown op {}", ceph_osd_op_name(op.op));
+      throw std::runtime_error(
+        fmt::format("op '{}' not supported", ceph_osd_op_name(op.op)));
+    }
+  }).handle_error_interruptible(
+    osd_op_errorator::all_same_way([&osd_op]
+      (auto e) -> OpsExecuter::osd_op_errorator::future<> {
+      if (osd_op.op.flags & CEPH_OSD_OP_FLAG_FAILOK &&
+        e.value() != EAGAIN && e.value() != EINPROGRESS) {
```

Comment:
For `EAGAIN` and `EINPROGRESS` we could leverage the `handle_error()`'s ability to accept multiple handlers (pseudocode):

```cpp
.handle_error(
  crimson::ct_error::eagain.handle([] {}),
  all_same_way([] {}));
```